python/cudf/cudf/core/reshape.py

```diff
@@ -269,7 +325,11 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
             # objs are empty dataframes.
             return cudf.DataFrame()
         elif len(objs) == 1:
-            if ignore_index:
+            if join == "inner":
+                result = cudf.DataFrame()
+                result.index = cudf.RangeIndex(len(objs[0]))
+                return result
```

Comment:
Can combine these instead of initializing then modifying:
```suggestion
                return cudf.DataFrame(index=cudf.RangeIndex(len(objs[0])))
```

Reply:
Great :D